runHarness/AppInstance/AuctionAppInstance.pm

```diff
@@ -111,21 +110,25 @@ override 'checkConfig' => sub {
 		$console_logger->error( "Workload $workloadNum, AppInstance $appInstanceNum: The Auction benchmark cannot be run with fewer than " . $minimumUsers . " users\n" );
 		return 0;
 	}
-
-	my $validWorkloadProfilesRef = $WeathervaneTypes::workloadProfiles->{"auction"};
-	if ( $self->getParamValue('workloadProfile') ~~ @$validWorkloadProfilesRef ) {
+	my $validWorkloadProfilesRef = $WeathervaneTypes::workloadProfiles{"auction"};
+    if (!(grep { $self->getParamValue('workloadProfile') eq $_ } @$validWorkloadProfilesRef)) {
 		$console_logger->error("Workload $workloadNum, AppInstance $appInstanceNum: The Workload-Profile for the Auction workload must be one of: @$validWorkloadProfilesRef");
 		return 0;
 	}
 
-	my $validAppInstanceSizes = $WeathervaneTypes::appInstanceSizes->{"auction"};
-	if ( $self->getParamValue('appInstanceSize') ~~ @$validAppInstanceSizes ) {
-		$console_logger->error("Workload $workloadNum, AppInstance $appInstanceNum: The AppInstance size for the Auction workload must be one of: @$validAppInstanceSizes");
-		return 0;
-	}
+    my $validAppInstanceSizes = $WeathervaneTypes::appInstanceSizes{"auction"};
+    if (!(grep { $self->getParamValue('configurationSize') eq $_ } @$validAppInstanceSizes)) {
```

Comment:
Question: why did you change the text matching method from '~~' to grep?

Reply:
I was having a strange issue with this test failing.  I thought that it might be due to some strange behavior of ~~, which is an experimental feature and is no longer really supported.  So I changed all of the ~~ matching in this file to use grep.  Then I realized that the test was using an otherwise unused parameter called _appInstanceSize_ rather than the correct _configurationSize_.  I must have used _appInstanceSize_ initially and then changed the name without deleting the parameter or changing it here.